File: FinanceDashboard.backend/FinanceDashboard.Application/DTOs/User/UserProfileDTO.cs

using FinanceDashboard.Application.DTOs.Category;

namespace FinanceDashboard.Application.DTOs.User
{
    public class UserProfileDTO
    {
        public string? UserName { get; set; } = string.Empty;
        public Decimal? MonthlyIncome { get; set; } = 0;
        public Decimal? MonthlySpendingAverage { get; set; } = 0;   
        public Decimal? AverageDailySpending { get; set; } = 0;
        public ICollection<CategoryListDTO>? Categories { get; set; } = new List<CategoryListDTO>();
        public string? Error { get; set; } = string.Empty;
        public bool IsSuccess { get; set; }   
    }
}

File: FinanceBoard.frontend/src/Models/DTOs/User/UserProfileDTO.ts
import type { CategoryList } from "../Category/CategoryList";

export type UserProfileDTO = {
    userName? : string;
    MonthlySpendingAverage? : number;
    MonthlyIncome? : number;
    AverageDailySpending? : number;
    exception? : string;
    Categories? : CategoryList[];
}